### PR TITLE
updated other gamepad button and axes states

### PR DESCRIPTION
**Pull request by satyapoojasama**
OpenXR provides paths to only few states of button, but webxr apps expect the other states to be populated too. In this PR, those states are updated, For more reference - https://www.w3.org/TR/gamepad/#gamepadbutton-interface

### CODE REVIEW

File: Dependencies/xr/Source/OpenXR/XR.cpp [hunk 1]
```diff
@@ -1480,6 +1480,15 @@ namespace xr
                     {
                         // Only signal that gamepad data is available if the actions were available
                         InputSources[idx].GamepadTrackedThisFrame = true;
```
Comment (rickfromwork) [inline Dependencies/xr/Source/OpenXR/XR.cpp:1482]: nit: mark gamepad as tracked after setting all the values

File: Dependencies/xr/Source/OpenXR/XR.cpp [hunk 2]
```diff
@@ -1480,6 +1480,15 @@ namespace xr
...
+                        gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Pressed = (gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value == 1);
+                        gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Touched = (gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value > 0);
+                        gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Touched = (gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Touched = (gamepadObject.Axes[controllerInfo.THUMBSTICK_X_AXIS] != 0 || gamepadObject.Axes[controllerInfo.THUMBSTICK_Y_AXIS] != 0);
```
Comment (rickfromwork) [inline Dependencies/xr/Source/OpenXR/XR.cpp:1491]: This one is interesting. It makes sense at a glance, but I wonder if it's what the end receivers of input are expecting. Do you have a documentation link for this?

File: Dependencies/xr/Source/OpenXR/XR.cpp (hunk 1 above)
Comment (satyapoojasama) [reply Dependencies/xr/Source/OpenXR/XR.cpp:1482]: Sure!

File: Dependencies/xr/Source/OpenXR/XR.cpp (hunk 2 above)
Comment (satyapoojasama) [reply Dependencies/xr/Source/OpenXR/XR.cpp:1488]: Yes, I found it here: https://github.com/immersive-web/webxr-input-profiles/tree/master/packages/motion-controllers#thumbstick-and-touchpad-components

**Review by rickfromwork — APPROVED** [review]

**Pull request merged**